### DISCUSSION TITLE
Guidance for HDR -> SDR

### DISCUSSION THREAD

**Discussion by workschoolhub** [iina/iina#4734] [Q&A]
I use a Studio Display and I’m trying to figure out the optimal settings for watching HDR videos on an SDR display. It’s been a bit of a struggle, so I thought I’d reach out here.

For example, it appears that I get good results when I check “Enable HDR mode” in the “Video/Audio” settings but only if I also adjust the IINA contrast in the Video Panel by about -30 to -20 (depending on how bright I’ve set my monitor).

I also notice that the tone mapping algorithm (e.g. reinhard) doesn’t really do anything perceivable.

Questions:
1. Should HDR mode be enabled for watching HDR videos on an SDR monitor?
2. Are there guidelines for optimizing IINA settings for SDR monitors?
3. Also, are there guides for using MPV settings in IINA? … for example, if I wanted to try the MPV “bt.2446a” tone-mapping algorithm in IINA?

Thanks all. I’m glad IINA exists. I couldn’t quite get my colors to “feel” right on VLC.

**Comment by low-batt** [comment]
I'm the wrong person to be answering this, but here are some thoughts…

On this:
> Should HDR mode be enabled for watching HDR videos on an SDR monitor?

When the IINA `Enable HDR support` setting is checked IINA will activate HDR support _if appropriate_.  IINA will check that it is playing a HDR video and that macOS indicates the screen supports [Extended Dynamic Range](https://developer.apple.com/documentation/metal/hdr_content/displaying_hdr_content_in_a_metal_layer#3335505) before activating HDR. Thus it is fine to have this setting checked when a SDR monitor is being used. This setting is there to give you the ability to disable HDR support.

On this:
> Also, are there guides for using MPV settings in IINA? … for example, if I wanted to try the MPV “bt.2446a” tone-mapping algorithm in IINA?

From the `mpv` manual entry for [tone-mapping](https://mpv.io/manual/stable/#options-tone-mapping):
> **bt.2446a**
    HDR<->SDR mapping specified in ITU-R Report BT.2446, method A. This is the recommended curve for well-mastered content. (`--vo=gpu-next only`)

This part is important:  (`--vo=gpu-next only`)

That refers to the [gpu-next](https://mpv.io/manual/stable/#video-output-drivers-gpu-next) video output driver. This is the new renderer that uses [Metal](https://developer.apple.com/metal/). This support is not available yet in `libmpv`. IINA must use the [OpenGL](https://www.opengl.org/) based renderer, which means `bt.2446a` can not be used.

Recently `mpv` has added a number of HDR features, but they can only be used with `gpu-next`.

The IINA wiki post [MPV Options and Properties](https://github.com/iina/iina/wiki/MPV-Options-and-Properties) provides some guidance on using `mpv` settings.

On this:
> Are there guidelines for optimizing IINA settings for SDR monitors?

I'll have to defer to others on this one.

**Reply by workschoolhub** [reply]
Many thanks for this response. It’s quite helpful. Hopefully others can chime in on optimizing settings for SDR monitors.

I’ve had a chance to play with other HDR clips, and it’s pretty confusing trying to get settings right; it may be that what works well for one video doesn’t work well for others.

It also seems the tone mapping algorithm doesn’t do anything (at least on my monitor).

Thanks again!

**Comment by low-batt** [comment]
The effects of tone mapping can be subtle. In [this comment](https://github.com/iina/iina/issues/4312#issuecomment-1521142414) I provide a test to run that exaggerates the effect to confirm it is working.

On the difference between videos…

There is more than one way to encode a HDR video. The industry has worked hard to develop many different [HDR formats](https://en.wikipedia.org/wiki/High-dynamic-range_television#Formats) for encoding videos. I guess in order to make things super complex and buggy to keep software engineers employed for years to come? Last I checked [mpv](https://github.com/mpv-player/mpv) has some issues with videos encoded in [Dolby Vision](https://en.wikipedia.org/wiki/Dolby_Vision).

Another factor is that authors can choose the dynamic range they want to use when encoding a video.

Not sure either of these account for your experience. These are just examples of some of the factors that might be in play.
